nimare/meta/utils.py

```diff
@@ -359,19 +363,33 @@ def compute_kda_ma(
         idx = (np.min(sphere, 1) >= 0) & (np.max(np.subtract(sphere, shape), 1) <= -1)
         sphere = sphere[idx, :].astype(int)
         exp = exp_idx[i]
-        if sum_overlap:
-            kernel_data[exp][tuple(sphere.T)] += value
+        if memmap_filename:
+            if sum_overlap:
+                kernel_data[exp][tuple(sphere.T)] += value
+            else:
+                kernel_data[exp][tuple(sphere.T)] = value
+            if i % chunk_size == 0:
+                kernel_data.flush()
         else:
-            kernel_data[exp][tuple(sphere.T)] = value
-
-        if memmap_filename and i % chunk_size == 0:
-            # Write changes to disk
-            kernel_data.flush()
+            if kernel_data_lst[exp] is None:
+                kernel_data = np.zeros(shape, dtype=type(value))
+            else:
+                kernel_data = kernel_data_lst[exp].todense()
+            if sum_overlap:
+                kernel_data[tuple(sphere.T)] += value
+            else:
+                kernel_data[tuple(sphere.T)] = value
 
-    if squeeze:
-        kernel_data = np.squeeze(kernel_data, axis=0)
+            kernel_data_lst[exp] = sparse._compressed.GCXS.from_numpy(
```

Comment:
`kernel_data` is a dense_matrix composed of zeros at this point right? 
 
I think those use `sparse` matrices correctly we want to avoid dense matrices at any point, and instead build a sprase matrix from the bottom up. 

See: https://sparse.pydata.org/en/stable/construct.html

I think we want to create from coordinates and data, rather than from an empty numpy matrix. This should save memory & processing time.

Reply:
Thanks for the review. I agree with this. Now, I can see how to implement it. 
I think in the `KernelTrnasformer` we do need to convert the sparse arrays to dense since `nib.Nifti1Image()` expects a dense array.

Reply:
Okay, yes, I think that's correct, although at that point we will have one less dimension (i.e. studies) so the dense array should have a much smaller footprint. 

Hopefully this speeds things up. If not I'm OK to try scipy.sparse also